homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -0,0 +1,234 @@
+    hass.async_create_task(hass.config_entries.async_reload(entry.entry_id))
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry, async_add_entities
+):
+    """Set up the electricity price sensor from config_entry."""
+    if not config_entry.update_listeners:
+        config_entry.add_update_listener(update_listener)
+
+    name = config_entry.data[CONF_NAME]
+    entity_id = async_generate_entity_id(ENTITY_ID_FORMAT, name, hass=hass)
+
+    pvpc_data_handler = PVPCData(
+        tariff=config_entry.options.get(
+            ATTR_TARIFF, config_entry.data.get(ATTR_TARIFF, DEFAULT_TARIFF)
+        ),
+        local_timezone=hass.config.time_zone,
+        websession=async_get_clientsession(hass),
+        logger=_LOGGER,
+        timeout=config_entry.data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+    )
+    async_add_entities([ElecPriceSensor(name, entity_id, pvpc_data_handler)], True)
+
+
+class ElecPriceSensor(RestoreEntity):
+    """Class to hold the prices of electricity as a sensor."""
+
+    unit_of_measurement = UNIT
+    icon = ICON
+    should_poll = False
+
+    def __init__(self, name, entity_id, pvpc_data_handler):
+        """Initialize the sensor object."""
+        self._name = name
+        self.entity_id = entity_id
+        self._pvpc_data = pvpc_data_handler
+        self._num_retries = 0
+
+        self._init_done = False
+        self._hourly_tracker = None
+        self._price_tracker = None
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cancel listeners for sensor updates."""
+        self._hourly_tracker()
+        self._price_tracker()
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        if state:
+            self._pvpc_data.state = state.state
+
+        # Update 'state' value in hour changes
+        self._hourly_tracker = async_track_time_change(
+            self.hass, self.async_update, second=[0], minute=[0]
+        )
+        # Update prices at random time, 2 times/hour (don't want to upset API)
+        random_minute = randint(1, 29)
+        mins_update = [random_minute, random_minute + 30]
+        self._price_tracker = async_track_time_change(
+            self.hass, self.async_update_prices, second=[0], minute=mins_update,
+        )
+        _LOGGER.debug(
+            "Setup of price sensor %s (%s) with tariff '%s', "
+            "updating prices each hour at %s min",
+            self.name,
+            self.entity_id,
+            self._pvpc_data.tariff,
+            mins_update,
+        )
+        await self.async_update_prices(dt_util.utcnow())
+        self._init_done = True
+        await self.async_update_ha_state(True)
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return "_".join([DOMAIN, "sensor", self.entity_id])
```

Comment:
unique_ids are per domain so you don't need DOMAIN here.

Can you come up with something that is more unique

Reply:
I think you can use the tariff here as well